csharp/src/Drivers/Databricks/BaseDatabricksReader.cs

```diff
@@ -42,7 +42,7 @@ protected BaseDatabricksReader(DatabricksStatement statement, Schema schema, boo
             this.schema = schema;
             this.isLz4Compressed = isLz4Compressed;
             this.statement = statement;
-            if (statement.DirectResults != null && !statement.DirectResults.ResultSet.HasMoreRows)
+            if (statement.DirectResults?.ResultSet?.HasMoreRows == true)
```

Comment:
@jackyhu-db Sorry. My mistake!
```suggestion
            if (statement.DirectResults?.ResultSet? != null && !statement.DirectResults.ResultSet.HasMoreRows)
```